packages/next/src/client/components/navigation.ts

```diff
@@ -243,7 +242,9 @@ export function useSelectedLayoutSegment(
     return null
   }
 
-  return selectedLayoutSegments[0]
+  return parallelRouteKey === 'children'
+    ? selectedLayoutSegments[0]
+    : selectedLayoutSegments[selectedLayoutSegments.length - 1] ?? null
```

Comment:
if this is `__DEFAULT__`, I feel like it would make more sense to return null rather than returning that internal string. Since `useSelectedLayoutSegment` is intended to return the active route, and default is intended as a fallback for when there isn't a route defined, it seems reasonable to assume that there is no active slot for the parallel route.